sled-agent/src/instance_manager.rs

```diff
@@ -67,11 +68,8 @@ impl InstanceManager {
         underlay_ip: Ipv6Addr,
         gateway_mac: MacAddr6,
     ) -> InstanceManager {
-        let data_link = crate::common::underlay::find_chelsio_links()
-            .unwrap()
-            .into_iter()
-            .next()
-            .unwrap();
+        let data_link =
+            underlay::find_chelsio_links().unwrap().into_iter().next().unwrap();
```

Comment:
I realize this didn't change in this PR (or mine where I moved it too), but looking at it again: I don't think we should be unwrapping here (in either case). On non-gimlets this is fine because `find_chelsio_links` doesn't actually do anything and just returns a fixed list, but on real gimlets `find_chelsio_links` can fail, or it can succeed while returning an empty vec.

If you don't want to fix it as part of this PR I can do it in a followup.

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/2469